src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java

```diff
@@ -459,6 +463,12 @@ private void sendData(MPDbAdapter dbAdapter, MPDbAdapter.Table table, String[] u
                         sendEmptyMessageDelayed(FLUSH_QUEUE, mTrackEngageRetryAfter);
                         mFailedRetries++;
                         logAboutMessageToMixpanel("Retrying this batch of events in " + mTrackEngageRetryAfter + " ms");
+                        break;
+                    }
+
+                    eventsData = dbAdapter.generateDataString(table);
+                    if (eventsData != null) {
+                        queueCount = Integer.valueOf(eventsData[2]);
```

Comment:
I know it's not your fault but can we give `2` a meaningful name? Everything single time I look at this code I have to figure out what `generateDataString` does.